diff --git a/uc_credit.module b/uc_credit.module
index 7eb1863..e7e926a 100644
--- a/uc_credit.module
+++ b/uc_credit.module
@@ -417,6 +417,18 @@ function uc_payment_method_credit($op, &$arg1, $silent = FALSE) {
         'cc_cvv' => check_plain($_POST['cc_cvv']),
         'cc_bank' => check_plain($_POST['cc_bank']),
       );
+      //---------------patch-------------------
+      $order = uc_order_load($_SESSION['cart_order']);
+      if( !empty($order->line_items) && is_array($order->line_items) ){
+        foreach ($order->line_items as $val){
+          if($val['line_item_id'] == 'subtotal'){
+            if ($val['amount'] == 0 && !empty($_SESSION['uc_discounts_codes']) ){
+              return true;
+            }
+          }
+        }
+      }
+      //---------------------------------------
       // Recover cached CC data in $_POST if it exists.
       if (isset($_POST['payment_details_data'])) {
         $cache = uc_credit_cache('save', $_POST['payment_details_data']);
